### PR TITLE
[Impeller] Gate Vulkan selection on API 29

### PR DESCRIPTION
**Pull request by dnfield**
Platform Views on Android require API 29 when using Vulkan.

~~I'm not sure if I did the CI changes right, want to run it here and see. The added test needs to run on a device or emulator with API 29 or above and API 28 or below to be valid.~~

Makes the following changes to CI:

- Runs the emulator tests on presubmit, but only if `shell/platform/android/**` or `lib/ui/**` or `.ci.yaml` changes.
- Changes the emulator tests to be a regular engine_v2 test that spawns an x86 build. Unfortunately, the older API level emulators are only available for x86, not x86_64, and you cannot run the binary we create unless it matches the ABI.
- Runs the new test on both API 28 and API 34.

Fixes https://github.com/flutter/flutter/issues/132984

/cc @johnmccutchan fyi

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -88,11 +88,15 @@ targets:
   - name: Linux linux_android_emulator_tests
     enabled_branches:
       - main
-    recipe: engine_v2/builder
-    presubmit: false
+    recipe: engine_v2/engine_v2
+    bringup: true
```
Comment (dnfield) [inline .ci.yaml:92]: This needs some time for the engine_v2 selection to propagate, per @godofredoc

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by matanlurey — APPROVED** [review]
LGTM!

**Comment by zanderso** [comment]
What's the plan for the scenario app tests? If they aren't running anymore should they be deleted? Should we stop building them at least?

**Pull request merged**